File: C#/7 kyu/0015.cs
// Kata: Return the first M multiples of N
// Link: https://www.codewars.com/kata/593c9175933500f33400003e

// Implement a function that takes two numbers m and n and returns an array of
// the first m multiples of the real number n. Assume that m is a positive
// integer.

// Ex.

// (3, 5.0) --> [5.0, 10.0, 15.0]

using System;
using System.Collections.Generic;

public class Kata
{
  public static double[] Multiples(int m, double n)
  {
    List<double> results = new List<double>();
    for (int i = 1; i<= m; ++i) {
        results.Add(n*i);
    }
    return results.ToArray();
  }
}

File: TypeScript/7 kyu/0015.ts
// Kata: Sum of Cubes
// Link: https://www.codewars.com/kata/59a8570b570190d313000037

// Write a function that takes a positive integer n, sums all the cubed values
// from 1 to n (inclusive), and returns that sum.

// Assume that the input n will always be a positive integer.

// Examples: (Input --> output)

// 2 --> 9 (sum of the cubes of 1 and 2 is 1 + 8)
// 3 --> 36 (sum of the cubes of 1, 2, and 3 is 1 + 8 + 27)

export function sumCubes(n: number): number {
  return Array(n)
    .fill(0)
    .map((_: number, i: number) => i + 1)
    .reduce((sum: number, current: number) => sum + current ** 3, 0);
}
